### PR TITLE
Adding documentation for Retry analyzer

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes https://github.com/cbeust/testng/issues/396

### CODE REVIEW

**Comment by juherr** [comment]
Strange that Travis failed.

**Comment by krmahadevan** [comment]
@juherr - But what are we building here 😁

**Comment by juherr** [comment]
🤷‍♂️

File: doc/documentation-main.html [hunk 1]
```diff
@@ -1793,6 +1793,52 @@ <h4><a class="section" indent=".." name="rerunning">Rerunning failed tests</a></
 methods so that you are guaranteed to run the methods that failed without any 
 SKIP failures.</p>
 
+Sometimes you might also want TestNG to automatically retry a test whenever it fails. In those situations you can make use of a retry analyzer.
```
Comment (cbeust) [inline doc/documentation-main.html]: Sometimes, you might want TestNG to automatically retry a test whenever it fails. In those situations, you can use a retry analyzer. When you bind a retry analyzer to a test, TestNG automatically invokes the retry analyzer to determine if TestNG can retry a test case again in an attempt to see if the test that just fails now passes.

Here is how you use a retry analyzer:

File: doc/documentation-main.html [hunk 2]
```diff
@@ -1793,6 +1793,52 @@ <h4><a class="section" indent=".." name="rerunning">Rerunning failed tests</a></
...
+When you bind a retry analyzer to a test, TestNG automatically invokes the retry analyzer to determine if TestNG can retry a test-case
+once again in an attempt to see if the flaky test passes. Following is how you can use a retry analyzer.
+
+<ol>
+  <li>Build an implementation of the interface <tt>org.testng.IRetryAnalyzer</tt></li>
+  <li>Bind this implementation to the <tt>@Test</tt> annotation for e.g., <tt>@Test(retryAnalyzer = LocalRetry.class)</tt></li>
+</ol>
+
+Following is a sample implementation of the retry analyzer that retries a test for a maximum of 3 times.
```
Comment (cbeust) [inline doc/documentation-main.html]: "three" times (spell out small numbers).

File: doc/documentation-main.html (hunk 1 above)
Comment (krmahadevan) [reply doc/documentation-main.html]: Amended the commit and fixed as suggested

File: doc/documentation-main.html (hunk 2 above)
Comment (krmahadevan) [reply doc/documentation-main.html]: Amended the commit and fixed as suggested

**Comment by krmahadevan** [comment]
ping @cbeust - I have fixed the review comments. Please help take a look

**Pull request merged**